pkg/virt-manifest/mapper.go

```diff
@@ -42,52 +43,40 @@ type savedDisk struct {
 	disk v1.Disk
 }
 
-func ExtractPvc(dom *v1.DomainSpec) (*v1.DomainSpec, []savedDisk) {
-	specCopy := &v1.DomainSpec{}
-	model.Copy(specCopy, dom)
-
+func ExtractPvc(dom *v1.DomainSpec) []savedDisk {
 	pvcDisks := []savedDisk{}
-	allDisks := []v1.Disk{}
 
-	for idx, disk := range specCopy.Devices.Disks {
+	for idx, disk := range dom.Devices.Disks {
 		if disk.Type == Type_PersistentVolumeClaim {
 			// Save the disk so we can fix it later
 			diskCopy := v1.Disk{}
 			model.Copy(&diskCopy, disk)
 			pvcDisks = append(pvcDisks, savedDisk{disk: diskCopy, idx: idx})
-
-			// Alter the disk record so that libvirt will accept it
-			disk.Type = Type_Network
-			disk.Source.Protocol = "iscsi"
 		}
-		allDisks = append(allDisks, disk)
 	}
-	// Replace the Domain's disks with modified records
-	specCopy.Devices.Disks = allDisks
 
-	return specCopy, pvcDisks
+	return pvcDisks
 }
 
 // This is a simplified version of the domain creation portion of SyncVM. This is intended primarily
 // for mapping the VM spec without starting a domain.
-func MapVM(con virtwrap.Connection, vm *v1.VM) (*v1.VM, error) {
+func MapVM(con virtwrap.Connection, vm *v1.VM, k8sClient kubernetes.Interface) (*v1.VM, error) {
```

Comment:
All defaults which you need to have a persistend spec from the VM OS view. The reasons are:

 * Having a more meaningful spec. It is awkward, that parts of the spec, unnecessarily come back via the status (e.g. PCI addresses). That forces you, for later re-runs, to merge your spec with a status, if you care about persistent specs.
 * Synchronize the meaning of a VM spec, with the meaning of specs in k8s. A spec there is complete when it is accepted by the system, and went through all admission controllers and initializers (except from the final scheduling decision, that  will asynchronously fill in spec.nodeName). That does not mean that every possible field in the Pod spec is filled out.
 * Give us more control over the status, by making deliberate decisions what we want to expose, and why. Just doing dump domain copies into the status, means  that we clutter the status with an immense blob of mostly uninteresting information.